code/API_definitions/qod-api.yaml

```diff
@@ -329,18 +402,236 @@ components:
+          allOf:
+            - $ref: "#/components/schemas/Rate"
+            - description: |
+              The maximum best effort data
+        maxUpstreamBurstRate:
+          allOf:
+            - $ref: "#/components/schemas/Rate"
+            - description: |
+              The burst rate for best effort data
+        targetMinDownstreamRate:
+          allOf:
+            - $ref: "#/components/schemas/Rate"
+            - description: |
+              This is the target minimum downstream rate for the QoS profile.
+
+              For 5G networks 3GPP Guaranteed Bit Rate (GBR) refers to a dedicated, fixed data rate assigned to
+              specific services, ensuring a minimum performance level. As per 3GPP TS 23.203,
+              GBR is a QoS parameter used to manage traffic classes in mobile networks. It
+              provides a stable data rate for latency-sensitive applications, such as voice calls or
+              video streaming, helping maintain a consistent user experience.  When this attribute is set
+              this could imply that a GBR QCI is used, though mechanisms on the network can be used to
+              ensure a minimum performance level without using a GBR QCI.
+
+              The committed data rate allocated to specific services, ensuring a consistent level of
+              performance. DOCSIS networks, the "Minimum Reserved Traffic Rate" is defined in the 
+              DOCSIS 3.1 - MAC and Upper Layer Protocols Interface Specification" 
+              and it ensures a consistent level of performance for specific services within the network.
+        maxDownstreamRate:
+          allOf:
+            - $ref: "#/components/schemas/Rate"
+            - description: |
+              The maximum best effort rate
+        maxDownstreamBurstRate:
+          allOf:
+            - $ref: "#/components/schemas/Rate"
+            - description: |
+              The maximum best effort burst rate
+        minDuration:
+          allOf:
+            - $ref: "#/components/schemas/Duration"
+            - description: |
+              The shortest time period that this profile can be deployed.
+        maxDuration:
+          allOf:
+            - $ref: "#/components/schemas/Duration"
+            - description: |
+              The maximum time period that this profile can be deployed.
+        priority:
+          type: number
+          example: 2.2
+          description: |
+            Priority levels allow efficient resource allocation and ensure optimal performance
+            for various services in each technology, with the highest priority traffic receiving
+            preferential treatment.
+            The lower value the higher priority.
+            Since not all access networks use the same priority range, so this priority will be
+            scaled to the access network's priority range.
+          format: float
+          minimum: 1
+          maximum: 10
+        packetDelayBudget:
```

Comment:
@eric-murray If we are going to constrain this to one-way, does that mean that we need to have an upStream and a downStream one?

Could we revert this to being a round-trip delay for the profile and then it's up to the operator to map this two up and down delays that fit within this budget?

@jlurien I'm happy to rename this to latency. 

Does anyone else have a comment on this attribute?

Reply:
Packet delay budget can be assumed the same for both uplink and downlink.

If you want to specify a round-trip delay, then you would need to rename, as otherwise some will assume that the 3GPP definition of packet delay budget applies without reading the definition.

Also, for reasons discussed in the last meeting, this could not be the end-to-end round-trip latency, as likely the server will be located outside the CSP's network, and the CSP cannot assure QoS over an external network. Also, the concept of round-trip latency to a point in the middle of the flow (the CSP Internet Gateway, for example) might be somewhat counter-intuitive, as no packets will be round-tripping from the IGW.

My preference is to keep packet delay budget as one-way latency from device to CSP IGW, and that number can be doubled if someone wants to compute the contribution of the CSP network to overall end-to-end latency. Keeping the name `packetDelayBudget` instead of `latency` will avoid developers thinking this is the end-to-end round-trip latency because they didn't read the documentation.

Reply:
US latency is usually longer than DS latency, so I don't think assuming that they are the same really works.

If we are set against defining round trip latency on the access network I'd prefer to split this to be `usPacketDelayBudget` and `dsPacketDelayBudget`.

Reply:
The *latencies* may be different, but the *budget* can be the same. This is the 3GPP definition:

> The Packet Delay Budget (PDB) defines an upper bound for the time that a packet may be delayed between the UE and the PCEF. For a certain QCI the value of the PDB is the same in uplink and downlink. The purpose of the PDB is to support the configuration of scheduling and link layer functions (e.g. the setting of scheduling priority weights and HARQ target operating points). The PDB shall be interpreted as a maximum delay with a confidence level of 98 percent.
> NOTE 1:	The PDB denotes a "soft upper bound" in the sense that an "expired" packet, e.g. a link layer SDU that has exceeded the PDB, does not need to be discarded (e.g. by RLC in E-UTRAN). The discarding (dropping) of packets is expected to be controlled by a queue management function, e.g. based on pre-configured dropping thresholds.

So in the 3GPP definition:
- Uplink and downlink PDB are the same value
- The PDB is not a maximum allowed latency, but the value above which the packet could be dropped (by schedulers), and will be considered dropped (for packet loss calculations). So for a specified packet error loss ratio of 1%, up to 1% of packets are allowed to exceed the specified PDB.

But fixed networks may require different values for uplink and downlink, so we may just need to live with 3GPP profiles needing to specify the same value for both.

Note that the latency between PCEF and the IGW will be minimal, so my preference would be to define PDB as the upper bound on latency between UE and IGW, which is the genuine "edge" of a CSP network.

Reply:
I agree that the term "packetDelayBudget" is more accurate for what we can offer. It can be left to the description to explain that it refers to the one-way latency between the device and the gateway from the operator's network to other networks, and that the E2E latency will be about two times this value plus the latency not controlled by the operator. 
The most relevant attributes to differentiate profiles are the assigned priority and rate intervals, while other parameters such as this one are rather derivative and informative.